app/Models/ForumTopicComment.php

```diff
@@ -39,6 +41,42 @@ protected static function newFactory(): ForumTopicCommentFactory
         return ForumTopicCommentFactory::new();
     }
 
+    // == search
+
+    public function toSearchableArray(): array
+    {
+        $body = $this->body;
+
+        // As of 2025-05-18, ~0.23% of payload bodies are extremely large (>20KB).
+        // We'll truncate those.
+        if (mb_strlen($body) > 20000) {
+            $body = mb_substr($body, 0, 20000);
+        }
```

Comment:
Would it be worth running this through the shortcode processor to strip out the non-visible stuff (like `[url=X]` or `[img=Y]`, `[b]`, etc)? The tradeoff is that the embedded records (games, achievements, etc) would get expanded to their titles.
![image](https://github.com/user-attachments/assets/530eaedb-3998-4fb7-957b-d35ebb3f12ca)
![image](https://github.com/user-attachments/assets/f4033ca9-6b3d-4349-9d1b-8af8b9df0036)

Reply:
I'm not sure at the moment we should do this -- the result models will still contain the non-transformed content, so the mismatch (unless handled gracefully by all consuming UIs) might be confusing or ambiguous.

Reply:
I think this applies to user IDs (`[user=1]`) as well, at least initially 🤔

If we use `stripAndClamp()` it implies anything referenced in the post will need to be reindexed when the content referenced in the post changes. This is complicated enough that I think we should probably hold for now on processing before indexing.